### ISSUE TITLE
Escaping "method" still does not work correctly

### ISSUE THREAD

**Issue by cknitt** [facebook/reason#2513]
With the latest BuckleScript master (BuckleScript 7.0.2-dev.1 @ b692f6b / Reason 3.5.1 @ 0b7f4e9), 

```reason
type request = {method: string};

let x = {method: "GET"};

let y = x.method;
```

gives me

```
  This expression has type request
  The field method does not belong to type request
Hint: Did you mean method_?```

**Comment by cristianoc** [comment]
@cknitt did you try `7.0.1`, i.e. the latest, instead of `7.0.0-dev.2`?

**Comment by cknitt** [comment]
I originally reported some issues for 7.0.1, see https://github.com/BuckleScript/bucklescript/issues/4008.

Bob asked me to retest with the latest master (7.0.2-dev.1, not 7.0.0-dev.2!), and it seems not everything is fixed yet.

**Comment by cristianoc** [comment]
@cknitt can't find `7.0.2-dev.1` anywhere. Anyhow the issue exists.

**Comment by cknitt** [comment]
@cristianoc `7.0.2-dev.1` is not released yet. It is the version number that's reported if you build the current BuckleScript master from source and run `bsb -version`.

**Comment by bobzhang** [comment]
I can confirm this is not fixed in 7.0.1 and master
